### PR TITLE
Fixes #750: Added Email Verification

### LINKED ISSUE #750: Email Verification
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behaviour:**
There is no email verification when user registered.

**Expected behaviour:**
There should be email verification

**Yes, I would Like to work**

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Fixes #750 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
- Added Email Verification

### CODE REVIEW

**Comment by yashLadha** [comment]
Have to do by Firebase.

File: backend/blueprint/api/helpers/verifyEmail.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import re
```
[the remark this answers is not in the document]
Comment (vaibhavsingh97) [reply backend/blueprint/api/helpers/verifyEmail.py]: @djmgit I don't know, how to fix this. can you please help me?

**Comment by vaibhavsingh97** [comment]
@yashLadha I do agree, but any validation done client-side can be disabled. So validation needs to exist in the backend as well unless we want to run the risk of incorrect or unsecure data being submitted through the form.

**Comment by yashLadha** [comment]
Why do we need to disable it in frontend? Can you give a practical scenario where it would be helpful because I never encountered a website that allows me even if I write a false email and send the request to the server?

**Comment by vaibhavsingh97** [comment]
@yashLadha You can always, you have to tweak source code. You can use `inspect element` and delete `type='email'` and if there is no server-side validation than user will be successfully registered

**Comment by yashLadha** [comment]
but before sending the data even if the user has given wrong input there will be some logic to check for the validation of data.

**Comment by vaibhavsingh97** [comment]
@yashLadha @gabru-md What if user disabled Javascript? Then Server-side validation comes to rescue. @djmgit Your opinion please

**Comment by gabru-md** [comment]
@vaibhavsingh97 If the user disables JS, then how will the UI even work. since all the build files are dependant on js?
@yashLadha am I right?

**Comment by yashLadha** [comment]
Yes the build files will be in JS and in what way you are thinking to disable JS?

**Comment by vaibhavsingh97** [comment]
@yashLadha @gabru-md After disabling js, the user will not able to use the application but the whole point is we should never rely on client-side validation.
Ref: https://stackoverflow.com/questions/43618459/should-i-duplicate-the-form-validation-on-client-side-in-server-side-just-like-c
https://stackoverflow.com/questions/15855770/why-do-we-need-both-client-side-and-server-side-validation
https://stackoverflow.com/questions/4902668/validating-form-input-client-side-js-vs-server-side
https://stackoverflow.com/questions/41052683/what-validation-is-better-to-practice-server-side-or-client-side-validation

**Comment by gabru-md** [comment]
Even if you are posting so many `links` here, I would still say that your backend implementation is also not proper. 
:confused:

**Comment by vaibhavsingh97** [comment]
@gabru-md you can always suggest, and I would update this PR with suggestions. Moreover, I would also working on pinging the SMTP server with the email address and see if it gets a success. if it's not then the email address is fake.

**Comment by yashLadha** [comment]
Have you seen my review comment? @vaibhavsingh97

**Comment by yashLadha** [comment]
You are misguiding the issue as all the validation details of email will be taken care by firebase 🔥

**Comment by djmgit** [comment]
@yashLadha @vaibhavsingh9 @gabru-md @ParthS007 Since we are going to integrate firebase for auth, and as it is pointed out earlier that firebase will be taking care of issues like email verification and all, is this issue still valid?
If not, I guess @vaibhavsingh97 can help with firebase integration and auth using firebase. @vaibhavsingh97 can you start working on firebase?

**Comment by vaibhavsingh97** [comment]
@djmgit you are right, after integrating firebase this issue not valid. Yes, I will start working on integration of firebase into badgeyay and will send PR soon

**Comment by harshit98** [comment]
Okay @vaibhavsingh97. Closing this PR.

**Pull request closed without merging**